### DISCUSSION TITLE
Register function does not return expected props

### DISCUSSION THREAD

**Discussion by hrasekj** [react-hook-form/react-hook-form#10596] [Q&A]
I have custom text field used like this:
```ts
<TextField
  error={errors.email?.message}
  label={t('summary.form.email')}
  {...register('email', {
    minLength: { value: 5, message: 'Email is too short' },
    required: true
  })}
/>
```

The TextField is implemented like this:
```ts
type TextFieldProps = {
  label: string;
  type?: HTMLInputTypeAttribute;
  error?: string;
  icon?: ReactNode;
} & Omit<UseFormRegisterReturn, 'ref'>;

const TextField = forwardRef<HTMLInputElement, TextFieldProps>(
  ({ label, type, error, icon, ...formRegisterProps }, forwardedRef) => {
    // ...

    return (
      <div>
        <input ref={forwardedRef} {...formRegisterProps} />
      </div>
    );
  }
);
```

When I saw type [UseFormRegisterReturn](https://github.com/react-hook-form/react-hook-form/blob/v7.45.0/src/types/form.ts#L166). My first thought was, when i pass props `required` or `minLength` to register function, they will be returned inside  `formRegisterProps` and can be used by the input element. But that is not happening. And I don't understand why. Do I missing something?

**Comment by leapful** [comment]
@hrasekj 

You need to enable `shouldUseNativeValidation` to apply native validation.

Reference: https://react-hook-form.com/docs/useform#shouldUseNativeValidation
